### ISSUE TITLE
"python.analysis.diagnosticMode" should be allowed even when "pyproject.toml" exists

### ISSUE THREAD

**Issue by Azureblade3808** [microsoft/pylance-release#6314]
## Environment data

-   Pylance version: v2024.8.103 (Pre-Release)
-   OS and version: Windows 10
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.12.5 Anaconda

## Code Snippet

N/A

## Repro Steps

1. Create or open a project with some Python files.
2. Add a "pyproject.toml" file with "tool.pyright" section.
3. Add a "python.analysis.diagnosticMode" item ".vscode/settings.json".

## Expected behavior

No error.

## Actual behavior

Error:

```plain
'python.analysis.diagnosticMode' cannot be set when a pyrightconfig.json or pyproject.toml is being used.
```

## Logs

N/A

**Comment by rchiodo** [comment]
Thanks for the issue but this by design. When a `pyproject.toml` is found, its settings override everything in your settings.json. That warning is to warn that your settings.json won't apply. 

The reasoning behind this is that all settings that generate diagnostics (or basically any Pyright only setting) shouldn't be overridden in your settings.json because your pyproject.toml is supposed to be the setting in use. It would be like overriding include file paths on a per user basis for a C++ project.

**Comment by erictraut** [comment]
@rchiodo, I agree with @Azureblade3808 that the `diagnosticMode` language server setting should not be affected by the absence or presence of a config file. It doesn't meet the requirements that you established when deciding which settings should be covered by this rule. In `diagnosticMode` has no corresponding setting the config file, it's specific to language servers, and it doesn't affect how types are evaluated or which errors are reported. It simply determines which errors show up in the "problems" pane of your editor.

**Comment by rchiodo** [comment]
Good point. I'll change it to exclude that one in particular. I thought it was also settable in pyrightconfig.json, but it seems not. Makes sense.

**Comment by debonte** [comment]
This issue has been fixed in prerelease version 2024.8.104, which we've just released. You can find the changelog here: [CHANGELOG.md](https://github.com/microsoft/pylance-release/blob/main/CHANGELOG.md#20248104-28-August-2024-PreRelease)
